edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeToDeviceMethodTest.cs

```diff
@@ -66,6 +66,17 @@ Task<MethodResponse> MethodHandler(MethodRequest methodRequest, object context)
                 await receiver.OpenAsync();
                 await receiver.SetMethodHandlerAsync("poke", MethodHandler, null);
 
+                var waitStart = DateTime.Now;
+                var isConnected = false;
+
+                while (!isConnected && (DateTime.Now - waitStart) < TimeSpan.FromSeconds(30))
+                {
+                    var connectedDevice = await rm.GetModuleAsync(edgeDeviceId, receiverModuleName);
+                    isConnected = connectedDevice.ConnectionState == DeviceConnectionState.Connected;
+                }
+
+                Assert.True(isConnected);
+
                 // Need longer sleep to ensure receiver is completely initialized
                 await Task.Delay(TimeSpan.FromSeconds(10));
```

Comment:
this no longer needed, right? :)

Reply:
Some waiting is still needed, because we need to make it sure that the subscription goes through. Probably it could be a lower value (as I noticed usually the connection took around 2 secs), I left it ten because tests tend to brake on too short waiting sometimes.